### PR TITLE
Profiling sign shreds on gpu

### PR DESCRIPTION
**Pull request by aeyakovenko**
#### Problem

@sakridge I am seeing slower performance with larger PinnedVecs, and slower perf on GPU vs non GPU.

#### Summary of Changes

Fixes #

tag: @sagar-solana

### CODE REVIEW

**Comment by sakridge** [comment]
> @sakridge I am seeing slower performance with larger PinnedVecs, and slower perf on GPU vs non GPU.

What do you mean larger PinnedVecs? How many packets are you signing at a time?

**Comment by aeyakovenko** [comment]
4096 packets per vec.

File: perf/src/recycler_cache.rs [hunk 1]
```diff
@@ -1,20 +1,28 @@
...
 pub struct RecyclerCache {
+    recycler_packets: PacketsRecycler,
     recycler_offsets: Recycler<TxOffset>,
     recycler_buffer: Recycler<PinnedVec<u8>>,
 }
 
 impl RecyclerCache {
     pub fn warmed() -> Self {
         Self {
+            recycler_packets: Recycler::warmed(1000, PACKETS_CAPACITY),
```
Comment (aeyakovenko) [inline perf/src/recycler_cache.rs]: @sakridge this is where I create the warmed pinned vec cache

**Comment by aeyakovenko** [comment]
@sakridge 32 packet sized entries i see
```
test bench_shredder_large_entries ... bench:  23,705,739 ns/iter (+/- 3,375,955)
```
1024 i see
```
bench_shredder_large_entries ... bench:  40,804,226 ns/iter (+/- 3,792,466
```
with GPUs on i see worse perf
32 packets per vec:
```
test bench_shredder_large_entries ... bench:  27,273,823 ns/iter (+/- 503,017)
```
1024 packets per vec with GPUs
```
test bench_shredder_large_entries ... bench:  94,859,799 ns/iter (+/- 4,801,893)
```
btw, GPUs are still failing on that 64th signature.

**Comment by sakridge** [comment]
> @sakridge 32 packet sized entries i see
> 
> ```
> test bench_shredder_large_entries ... bench:  38,343,341 ns/iter (+/- 7,743,477)
> ```
> 
> 1024 i see
> 
> ```
> test bench_shredder_large_entries ... bench:  64,834,141 ns/iter (+/- 5,296,322)
> ```
> 
> with GPUs on i see worse perf
> 32 packets per vec:
> 
> ```
> test bench_shredder_large_entries ... bench:  67,131,102 ns/iter (+/- 5,627,782)
> ```
> 
> 1024 packets per vec with GPUs
> 
> ```
> test bench_shredder_large_entries ... bench:  94,859,799 ns/iter (+/- 4,801,893)
> ```
> 
> btw, GPUs are still failing on that 64th signature.

Ok. Is this the PR to look at or the other one? I'll probably look at the 64th sig failure first.

**Pull request closed without merging**